monai/transforms/spatial/array.py

```diff
@@ -2609,3 +2619,168 @@ def _get_params(
+            See also: https://numpy.org/doc/stable/reference/generated/numpy.pad.html
+        pad_opts: padding options, see `numpy.pad`
+
+    """
+
+    backend = [TransformBackends.TORCH, TransformBackends.NUMPY]
+
+    def __init__(
+        self,
+        patch_size: Sequence[int],
+        start_pos: Sequence[int] = (),
+        fix_num_patches: Optional[int] = None,
+        overlap: Union[Sequence[float], float] = 0.0,
+        sort_key: Optional[Union[Callable, str]] = None,
+        pad_mode: Union[NumpyPadMode, str] = NumpyPadMode.CONSTANT,
+        pad_opts: Optional[Dict] = None,
+    ):
+        self.patch_size = ensure_tuple(patch_size)
+        self.start_pos = ensure_tuple(start_pos)
+        self.pad_mode: NumpyPadMode = look_up_option(pad_mode, NumpyPadMode)
+        self.pad_opts = {} if pad_opts is None else pad_opts
+        self.overlap = overlap
+        self.num_patches = 0
+        self.fix_num_patches = fix_num_patches
+        self.sort_key: Optional[Callable]
+        if isinstance(sort_key, str):
+            if sort_key == "random":
+                self.sort_key = np.random.random
+            if sort_key == "min":
+                self.sort_key = self._get_patch_sum
+            if sort_key == "max":
+                self.sort_key = self._get_negative_patch_sum
+            else:
+                ValueError(f'sort_key should be either "min", "max", or "random", "{sort_key}" was given.')
+        else:
+            self.sort_key = sort_key
+
+    @staticmethod
+    def _get_patch_sum(x):
+        return x[0].sum()
+
+    @staticmethod
+    def _get_negative_patch_sum(x):
+        return -x[0].sum()
+
+    def __call__(self, array: NdarrayOrTensor):
+        # create the patch iterator which sweeps the image row-by-row
+        array_np, *_ = convert_data_type(array, np.ndarray)
+        patch_iterator = iter_patch(
+            array_np,
+            patch_size=(None,) + self.patch_size,  # expand to have the channel dim
+            start_pos=(0,) + self.start_pos,  # expand to have the channel dim
+            overlap=self.overlap,
+            copy_back=False,
+            mode=self.pad_mode,
+            **self.pad_opts,
+        )
+        if self.sort_key is not None:
+            output = sorted(patch_iterator, key=self.sort_key)
+        else:
+            output = list(patch_iterator)
+        if self.fix_num_patches:
+            output = output[: self.fix_num_patches]
+            if len(output) < self.fix_num_patches:
+                if self.pad_opts.get("constant_values"):
+                    patch = np.ones((array.shape[0], *self.patch_size)) * self.pad_opts["constant_values"]
+                else:
+                    patch = np.zeros((array.shape[0], *self.patch_size))
+                location = np.zeros((3, len(self.patch_size)))
+                output += [(patch, location)] * (self.fix_num_patches - len(output))
+
+        if not isinstance(array, np.ndarray):
```

Comment:
No need to check whether it's `np.ndarray`, `convert_to_dst_type` will not convert if already the same type.

Thanks.